### PR TITLE
Issue #2222: Fixed primary navigation menu block not rendering as dro…

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/2222

### CODE REVIEW

**Comment by klonos** [comment]
@jenlampton that failure on php7 seems to be a random one. Can you please close and reopen this PR in order to restart the test? Other than that seems to work just fine 👍

File: core/modules/layout/layout.install [hunk 1]
```diff
@@ -35,6 +35,21 @@ function layout_install() {
   db_query("UPDATE {system} SET weight = 60 WHERE name = 'layout' AND type = 'module'");
 }
 
+/**
+ * Set home page menu to render as a drop-down menu.
+ */
+function layout_update_1000() {
+  $layout = layout_load('home');
```
Comment (quicksketch) [inline core/modules/layout/layout.install]: Let's add a check to see if the layout is still there in the first place. `layout_load()` will return FALSE if the given layout does not exist.

File: core/modules/layout/layout.install [hunk 2]
```diff
@@ -35,6 +35,21 @@ function layout_install() {
   db_query("UPDATE {system} SET weight = 60 WHERE name = 'layout' AND type = 'module'");
 }
 
+/**
+ * Set home page menu to render as a drop-down menu.
+ */
+function layout_update_1000() {
```
Comment (quicksketch) [inline core/modules/layout/layout.install]: Let's try putting this update in `standard.install`, as it actually "owns" the layout configuration. Being in layout.install is also correctly breaking the upgrade tests.

File: core/modules/layout/layout.install [hunk 3]
```diff
@@ -35,6 +35,21 @@ function layout_install() {
...
+  $layout = layout_load('home');
+  foreach ($layout->content as $uuid => $block_info) {
+    if ($block_info->delta == 'main-menu') {
```
Comment (quicksketch) [inline core/modules/layout/layout.install]: Maybe as an extra safe check `&& empty($layout->content[$uuid]->settings['block_settings']['style'])`

**Review by quicksketch — CHANGES_REQUESTED** [review]

File: core/modules/layout/layout.install (hunk 2 above)
Comment (jenlampton) [inline core/modules/layout/layout.install]: done.

File: core/modules/layout/layout.install (hunk 1 above)
Comment (jenlampton) [reply core/modules/layout/layout.install]: easy enough.

File: core/modules/layout/layout.install (hunk 3 above)
Comment (jenlampton) [reply core/modules/layout/layout.install]: it's not empty, it's `default`

File: core/modules/layout/layout.install (hunk 2 above)
Comment (jenlampton) [reply core/modules/layout/layout.install]: Nope, it won't run there.

Comment (jenlampton) [reply core/modules/layout/layout.install]: oh, it helps if you are on the right branch. Yes, it runs there!

File: core/modules/layout/layout.install (hunk 3 above)
Comment (jenlampton) [reply core/modules/layout/layout.install]: double checked: on a fresh install, it is empty.

File: profiles/standard/standard.install [hunk 4]
```diff
@@ -401,3 +401,20 @@ function standard_install() {
...
   config_set('system.core', 'cache', '1');
 }
+
+/**
+ * Set home page menu to render as a drop-down menu.
+ */
+function standard_update_1000() {
+  if ($layout = layout_load('home')) {
+    foreach ($layout->content as $uuid => $block_info) {
+      dvm($layout->content[$uuid]->settings['block_settings']['style']);
```
Comment (quicksketch) [inline profiles/standard/standard.install]: Remove `dvm()`. :wink:

**Review by quicksketch — CHANGES_REQUESTED** [review]

File: profiles/standard/standard.install [hunk 5]
```diff
@@ -401,3 +401,20 @@ function standard_install() {
   config_set('system.core', 'preprocess_js', '1');
   config_set('system.core', 'cache', '1');
 }
+
+/**
+ * Set home page menu to render as a drop-down menu.
+ */
+function standard_update_1000() {
+  if ($layout = layout_load('home')) {
```
Comment (quicksketch) [inline profiles/standard/standard.install]: Sorry I should have remembered this in the first round of feedback, but we shouldn't be using APIs like `layout_load()` in the update hook. Instead you should use config and load the file directly, e.g. `$config = config('layout.layout.home');` and then check `$config->isNew()` to see if it existed.

File: profiles/standard/standard.install (hunk 4 above)
Comment (jenlampton) [reply profiles/standard/standard.install]: pushed.

File: profiles/standard/standard.install (hunk 5 above)
Comment (jenlampton) [reply profiles/standard/standard.install]: okiedokie.

Comment (jenlampton) [inline profiles/standard/standard.install]: ugh, this stupid $config object is impossible to work with... are you sure I can't use $layout?

Comment (jenlampton) [inline profiles/standard/standard.install]: ...think I got it.

**Comment by quicksketch** [comment]
Thanks everyone! Merged into 1.x for 1.5.1.

**Pull request merged**